pycbc/population/scale_injections.py

```diff
@@ -0,0 +1,796 @@
+import numpy as np
+import glob, sys, copy, h5py
+from scipy import interpolate
+from scipy.integrate import quad
+from scipy.stats import kstest, ks_2samp
+from astropy.cosmology import WMAP9 as cosmo
+
+_mch_BNS = 1.4/2**.2
+_redshifts, _d_lum, _I = np.arange(0., 5., 0.01), [], []
+_save_params = ['mass1', 'mass2', 'spin1z', 'spin2z', 'spin1y', 'spin2y', 'spin1x', 'spin2x', 'distance', 'end_time']
+
+for zz in _redshifts:
+    _d_lum.append(cosmo.luminosity_distance(zz).value)
+_dlum_interp = interpolate.interp1d(_d_lum, _redshifts)
+
+def read_injections(folder_name):
+    ''' Read all the injections from the files in the provided folder.
+        The files must belong to individual set i.e. no files that combine 
+        all the injections in a run.
+        Identify injection strategies and finds parameter boundaries.
+        Collect injection according to GPS.
+
+        Parameters
+        ----------
+        folder_name: string
+           Location of folder containing simulation files
+
+        Returns
+        -------
+        chunk_data: dictionary
+           Contains the organized information about the injections
+    '''
+    
+    injections = {}
+    all_files = glob.glob(folder_name)
+    
+    nf, min_d, max_d = len(all_files), 1e12, 0
+    for i in range(nf):
+        injections[str(i)] = process_injections(all_files[i])
+        injections[str(i)]['file_name'] = all_files[i]
+        
+        mass1, mass2 = injections[str(i)]['mass1'], injections[str(i)]['mass2']   
+        mchirp, eta = m1m2_to_mcheta(mass1, mass2)
+        injections[str(i)]['chirp_mass'] = mchirp
+        injections[str(i)]['total_mass'] = mass1 + mass2
+    
+        min_m1, max_m1 = min(mass1), max(mass1)
+        min_m2, max_m2 = min(mass2), max(mass2)
+        min_totM, max_totM = min(mass1+mass2), max(mass1+mass2)        
+     
+        # Identify the mass distribution
+    
+        def cdf_componentMass(xx): #Assumes both the masses have the same distribution
+            return (xx - min_m2)/(max_m2 - min_m2)
+
+        def cdf_totalMass(xx):
+            return (xx - min_totM)/(max_totM - min_totM)
+ 
+        def cdf_log(xx): #Assumes both the masses have the same distribution
+            return (np.log(xx) - np.log(min_m2))/(np.log(max_m2) - np.log(min_m2))        
+    
+        p_thr = 0.0001
+        stat, pvalue = kstest(mass1, cdf_componentMass)
```

Comment:
Why are you doing a statistical test to determine which distribution it was? Why not simply take this as a command line option? (or keyword to the explicit functions).

Reply:
This was to make things least dependent on user. @tdent we need all the BBH files in all the folders, the delimiter can find files in multiple folders e.g. '*BBH*/*hdfinj*.hdf" and the statistical test will identify the injection distribution with nice accuracy. Alternatively we can specify every single BBH file and the corresponding injection strategy used.

Reply:
*BBH*/*hdfinj*.hdf